compiler/src/main/kotlin/dev/zacsweers/metro/compiler/fir/generators/ContributedInterfaceSupertypeGenerator.kt

```diff
@@ -67,9 +69,13 @@ internal class ContributedInterfaceSupertypeGenerator(
   }
 
   private val contributingTypesPredicate =
-    LookupPredicate.create {
-      annotated(classIds.allContributesAnnotations.map { it.asSingleFqName() })
+    LookupPredicate.create { annotated(classIds.allContributesAnnotations.asFqNames()) }
+
+  private val qualifiersPredicate by unsafeLazy {
+    DeclarationPredicate.create {
+      metaAnnotated(classIds.qualifierAnnotations.asFqNames(), includeItself = false)
     }
+  }
```

Comment:
Kind of nit: predicates are cheap. And don't even need to be part of the extension itself. So (1),don't need `unsafeLazy` and (2), consider putting all the predicates in a shared class like the `ClassId`s are now.

Reply:
This is the best type of nit 😁, I wasn't sure how expensive the predicates were before so it's helpful context going forward. Updating to move all the predicates used here to a unified file, and can put up a followup PR to migrate the other extensions to use it as well